Time gage rocover

ki43III opened the pull request:
## 課題
- 時計UI関連の三つだったか確かそれくらいのイシュー

## 概要

- 回復アイテムの効果を得られるようにする
- 回復アイテムの再出現を二種のパターン
- どちらのパターンも出現についてインスペクターから調整できる
- リスポーン時のタイム本数を設定　調整も出来る
- タイマー接触時のエフェクト　と時計UIにもエフェクト

## 変更点
変更点を挙げると長くなります　使用する際の注意点だけ書いておきます
使用する際は以下の写真のインスペクターの説明を読んでください
<img width="1440" alt="スクリーンショット 2024-02-23 2 56 12" src="https://github.com/CC-Circle/Janaihou/assets/133644847/b13a4ccb-4488-419f-8766-47fbb8e35197">
<img width="1440" alt="スクリーンショット 2024-02-23 2 56 29" src="https://github.com/CC-Circle/Janaihou/assets/133644847/734850ab-588e-47d3-bf15-d76f116695a7">
<img width="1440" alt="スクリーンショット 2024-02-23 2 56 21" src="https://github.com/CC-Circle/Janaihou/assets/133644847/49f5db6e-6864-4396-9d7c-4e76f88ca255">


## 影響範囲
上の写真に載っているクラスの変更が主ですね
シーンはステージ位置をコピーした物をtime_ricoverのような名前にして作成しました


## チェックリスト
<img width="1440" alt="スクリーンショット 2024-02-23 3 06 35" src="https://github.com/CC-Circle/Janaihou/assets/133644847/b5a898be-275d-4b52-bcf1-8dc62392ef73">
コンパイルはできました

ki43III committed:
時計オブジェクト追加

ki43III committed:
clockの移動

ki43III committed:
エフェクト以外の実装完了

ki43III committed:
フォルダ移動

ki43III committed:
etc

ki43III committed:
エフェクトを追加しました

ki43III committed:
フォルダの移動や説明文の追加など

ki43III committed:
不要なオブジェクト削除

ki43III committed:
不要なコード削除

Nibosi0501 commented:
Scene: time_recoverを開いた段階で、以下のエラーが発生しました。
```cmd
Problem detected while opening the Scene file: 'Assets/Scenes/time_recover.unity'.
Check the following logs for more details.
UnityEditor.EditorApplication:Internal_RestoreLastOpenedScenes ()

Prefab instance problem: 'vintage_pocket_watch_4k (Missing Prefab with guid: 9dd39d0f08199463883a08b7cfb37807)'. 4 instances are missing the same Prefab Asset with guid 9dd39d0f08199463883a08b7cfb37807.
UnityEditor.EditorApplication:Internal_RestoreLastOpenedScenes ()
```

詳細は、discordのスレッド「item: 逆転時計の回復アイテム」をみてください。

ki43III committed:
プレハブ読み込みエラー解決　その①プレハブ化のフォルダ構造を見直す

ki43III committed:
レハブ読み込みエラー解決　その②もう一回インポートしてみる

プレハブエラーが表示されなければ一歩前進ですね

ki43III committed:
諸々のスクリプトの適用を設定しました　これで動作する状態に戻ったと思います

ki43III committed:
逆転中にリスポーンした時は逆転を解除する処理を追加

ki43III committed:
ゲージ枯渇後のリスポーン

ki43III committed:
パラメータ調整とインスペクター表示調整

ki43III committed:
インスペクターの値の変更問題に対処

あの問題に遭遇
https://forum.unity.com/threads/custom-editor-problem-in-unity-5-3.372970/

ki43III committed:
フラク修正版

名前を間違えた

Nibosi0501 reviewed (approved):
ok
